### PR TITLE
Automated logs reporting for pytest >= 3.3

### PR DESCRIPTION
**Pull request by s0undt3ch**

### CODE REVIEW

**Comment by krasoffski** [comment]
Thank you!
Ow, this will require some time to read about new features of `pytest >= 3.3` and than review.

**Comment by s0undt3ch** [comment]
The 3.3 requirement is because the catchlog plugin was merged into pytest.

**Comment by s0undt3ch** [comment]
If we provide a specific fixture to allow attachements reporting, then we don't need to either overwrite the logging logger class  or patch the current logging logger class, thoughts?

**Comment by krasoffski** [comment]
@s0undt3ch 
It was similar fixture. This is good point to think for us because personally I like idea to avoid "hacking" logging class. Instead just use approach like:
```python 
from pytest_reportportal import attachment
```

On the other side, RP provides ability to add  descriptive message and log level to attachment (as I remember). In this case it might look like as on example:
```python
attach.info('CPU usage on %s', hostname, name='free_memory.txt', data=free_memory,
    mime='application/octet-stream')
```
This is quite similar to standard logging - that why I decided to "hack" logging. This, the idea looks promising we are not going to attach message and log level to attachment.

**Comment by s0undt3ch** [comment]
So, `attach` or `attachment` does not need to be imported(although it could be importable).
We make it a pytest fixture. Any tests that DO want to send attachments, they just add the fixture name to the test function and voila! Kudos to pytest for that.

If you are not following what I'm saying, no problem, I can make a second PR and I'll answer any doubts, concerns there.

About the semantics, it can be whatever we want it and accept whichever arguments we need.

**Comment by krasoffski** [comment]
> if you are not following what I'm saying, no problem, I can make a second PR and I'll answer any doubts, concerns there.
Oh, I forgot about this. 
If you interested in implementation of this fixture, It would very helpful from you side. Only one request to do not remove log patching for now.

File: pytest_reportportal/plugin.py [hunk 1]
```diff
@@ -39,7 +46,14 @@ def pytest_configure(config):
             "pytest report portal is not compatible with 'xdist' plugin.")
 
     # set Pytest_Reporter and configure it
-    config._reporter = RPReportListener()
+
+    if PYTEST_HAS_LOGGING_PLUGIN:
+        # This check can go away once we support pytest >= 3.3
+        config._reporter = RPReportListener(
+            _pytest.logging.get_actual_log_level(config, 'rp_log_level')
```
Comment (krasoffski) [inline pytest_reportportal/plugin.py]: When log level is not specified, `_pytest.logging.get_actual_log_level(config, 'rp_log_level')` returns `None` and as result:
```
INTERNALERROR> TypeError: Level not an integer or a valid string: None
```

File: pytest_reportportal/rp_logging.py [hunk 2]
```diff
@@ -70,3 +83,50 @@ def emit(self, record):
...
+    original_log = logger_class._log
+    original_makeRecord = logger_class.makeRecord
+
+    try:
+        def wrap_log(original_func):
+            @wraps(original_func)
+            def _log(self, *args, **kwargs):
+                attachment = kwargs.pop('attachment', None)
+                if attachment is not None:
+                    kwargs.setdefault('extra', {}).update({'attachment': attachment})
```
Comment (krasoffski) [inline pytest_reportportal/rp_logging.py]: Use max length 79 chars inclusive (85 > 79 characters)

File: pytest_reportportal/rp_logging.py [hunk 3]
```diff
@@ -70,3 +83,50 @@ def emit(self, record):
...
+            @wraps(original_func)
+            def makeRecord(self, name, level, fn, lno, msg, args, exc_info,
+                           func=None, extra=None, sinfo=None):
+                if extra is not None:
+                    attachment = extra.pop('attachment', None)
+                else:
+                    attachment = None
+                try:
+                    # Python 3.5
+                    record = original_func(self, name, level, fn, lno, msg, args,
```
Comment (krasoffski) [inline pytest_reportportal/rp_logging.py]: 85 > 79 characters

File: pytest_reportportal/rp_logging.py [hunk 4]
```diff
@@ -70,3 +83,50 @@ def emit(self, record):
...
+                                           exc_info, func=func, extra=extra,
+                                           sinfo=sinfo)
+                except TypeError:
+                    # Python 2.7
+                    record = original_func(self, name, level, fn, lno, msg, args,
```
Comment (krasoffski) [inline pytest_reportportal/rp_logging.py]: 81 > 79 characters

**Comment by s0undt3ch** [comment]
Review comments addressed.

**Comment by krasoffski** [comment]
@s0undt3ch many thanks! I will review again soon.

**Comment by s0undt3ch** [comment]
:+1:

**Comment by krasoffski** [comment]
@s0undt3ch, Could you please solve conflict and I will merge your PR?

**Comment by s0undt3ch** [comment]
Resolving conflicts online was successful, I'll fix it in a bit.

**Comment by s0undt3ch** [comment]
Not successful

**Comment by s0undt3ch** [comment]
Conflicts resolved.

**Comment by s0undt3ch** [comment]
:tada:

**Pull request merged**